### PR TITLE
Preetha/port customer release to 1.15

### PR DESCRIPTION
**Pull request by preetha-intel**
Port custom release code changes to 1.15 release

 Has new OV API compile_model() that improves FIL and memory consumption significantly.

### CODE REVIEW

**Review by sfatimar — COMMENTED** [review]
Sounds good, I do not see changes in MSFT graph partitioning code where we sort the initializers ?

File: onnxruntime/core/providers/openvino/backend_utils.cc [hunk 1]
```diff
@@ -57,7 +57,8 @@ CreateOVModel(const ONNX_NAMESPACE::ModelProto& model_proto, const GlobalContext
   const std::string model = model_proto.SerializeAsString();
   try {
     auto cnn_network = global_context.ie_core.ReadModel(model);
-    if (subgraph_context.precision == InferenceEngine::Precision::FP16) {
+    if ((subgraph_context.precision == InferenceEngine::Precision::FP16) &&
```
Comment (MaajidKhan) [inline onnxruntime/core/providers/openvino/backend_utils.cc:60]: Didn't HEMA add a condition here, that even if it's CPU and FP16, we need to apply ov::pass::ConvertFP32ToFP16 transformations? and that fixed issues when OVEP is run with CPU_FP16

**Comment by preetha-intel** [comment]
> Sounds good, I do not see changes in MSFT graph partitioning code where we sort the initializers ?

Thats already part of MSFT Master.
 https://github.com/microsoft/onnxruntime/pull/14631

**Comment by sfatimar** [comment]
LGTM

File: onnxruntime/core/providers/openvino/backend_utils.cc (hunk 1 above)
Comment (preetha-intel) [reply onnxruntime/core/providers/openvino/backend_utils.cc]: Done

**Pull request merged**